[android][image-picker] fix inconsistent naming of filesize property (filesize vs fileSize)

WookieFPV opened the pull request:
rename filesize to fileSize.
Android Implementation was different from iOS, ts definitions (ImagePicker.types.ts) & docu.
This was probably not intended and a bug


# Why

The naming of filesize was different on iOS (file**S**ize) and Android (file**s**ize).
This PR should unify the naming.
Support for fileSize in Android was added in #24524 but this improves the naming to make it the same on iOS & Android.

Naming:
file**s**ize: Android
file**S**ize: iOS, typescript, docs


# How

Replace in Files "filesize" -> "fileSize" in the whole packages/expo-image-picker/android folder.

# Test Plan

I did not test this feature.
This change is fairly simple.
But consumer that rely on the "wrong" field name might get a breaking change if this bug is fixed

# Checklist


- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

WookieFPV committed:
[android][image-picker] rename filesize to fileSize. Android Implementation was different from iOS, ts definitions (ImagePicker.types.ts) & docu.

alanjhughes reviewed (approved):
Thank you! Congrats on your first contribution! 🔥

WookieFPV committed:
[android][image-picker] add changelog entry for #27293.
I am not sure if this change should be marked as breaking.
It is breaking but it fixes a wrong behaviour and this makes the implementation match the docs

MarkSFrancis commented:
This change just caught us out too, thanks @WookieFPV! 
We'll keep an eye on the release notes to make sure we change our code from `filesize` for Android to `fileSize` once this is released.

Adriano2108 commented:
Is this merged into SDK 51, i am using SDK 50 and i still have this issue

entiendoNull commented:
Fixes https://github.com/expo/expo/issues/30895
